GVRf/Extensions/ResonanceAudio/resonanceaudio/src/main/java/org/gearvrf/resonanceaudio/GVRAudioListener.java

```diff
@@ -0,0 +1,294 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.resonanceaudio;
+
+import com.google.vr.sdk.audio.GvrAudioEngine;
+
+import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRComponent;
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVREventListeners;
+import org.gearvrf.GVRScene;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRTransform;
+import org.joml.Quaternionf;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a listener in the scene. It records the sound around it and plays through the
+ * speakers. It is allowed to have only one listener in the scene.
+ */
+public class GVRAudioListener extends GVRComponent {
```

Comment:
Hi @liaxim, I know that makes the most sense to attach the listener to the Main Camera but we could make some improvements to allow attach the listener to other scene objects like Unity3D does (https://docs.unity3d.com/Manual/class-AudioListener.html) so every scene has its own listener.

We didn't choose by some GVRAudioSession because we was going handle the states(paused/stopped/playing ...) of all audio sources of all scenes in the single GVRAudioSession adding some complexity to our logic.

But we can talk more about it and make improvements :)